app/controllers/user_data_controller.rb

```diff
@@ -2,43 +2,35 @@ class UserDataController < ApplicationController
   before_action :verify_jwt_subject!
 
   def show
-    @user_data = UserData.find_by!(record_retrieval_params)
+    user_data = UserData.find_by!(record_retrieval_params)
 
-    render json: ::UserDataPresenter.new(@user_data), status: :ok
-  end
-
-  def verify_jwt_subject!
-    unless @jwt_payload['sub'] == record_retrieval_params[:user_identifier]
-      raise Concerns::JWTAuthentication::SubjectMismatchError
-    end
+    render json: ::UserDataPresenter.new(user_data), status: :ok
   end
 
   # To keep things simple and fast on the client, we'll transparently
   # handle create or update in one method, called via POST
   def create_or_update
-    @user_data = find_or_build(record_retrieval_params)
-    success_status = @user_data.persisted? ? :ok : :created
+    user_data = UserData.find_or_initialize_by(record_retrieval_params)
+    success_status = user_data.persisted? ? :ok : :created
 
-    @user_data.payload = user_data_params[:payload]
-    @user_data.save!
+    user_data.payload = params[:payload]
+    user_data.save!
 
     render json: {}, status: success_status, format: :json
   end
 
   private
 
-  def find_or_build(attributes)
-    UserData.where(attributes).first || UserData.new(attributes)
+  def verify_jwt_subject!
```

Comment:
Wasn't this done in the module previously? What is different with this refactor?

Reply:
Discussed on slack, all good 👍